lib/new_relic/agent/span_event_primitive.rb

```diff
@@ -165,18 +165,18 @@ def custom_attributes segment
         end
       end
 
-      def merge_and_freeze_attributes agent_attributes, error_attributes
-        return agent_attributes.freeze unless error_attributes
-        return error_attributes.freeze if agent_attributes.equal?(NewRelic::EMPTY_HASH)
-        agent_attributes.merge!(error_attributes).freeze
+      def merge_and_freeze_attributes *hashes
+        hashes.each_with_object({}) do |hash, merged|
+          merged.merge!(hash) if hash
+        end.freeze
```

Comment:
I'm curious how this behaves when there is only one hash passed in in `hashes`? I'm thinking that maybe if there is just one that it might create extra allocations vs having an early return like we did before? I'm not sure how it works though.

Reply:
This is a good callout. I had a version that retained the early return functionality, but it became really complicated with the number of possible combinations after a third hash was tossed into the mix. Still, this method is only called in exactly one place, and we should strive to maintain the current level of object allocation. I'll put a TODO here and make sure this is addressed before it gets merged.

Reply:
Addressed with b86f528